File: backend/src/main/java/com/runboo/domain/tier/entity/Tier.java
package com.runboo.domain.tier.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "tier")
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Tier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "tier_id")
    private Long tierId;

    @Column(name = "name")
    private String name;

    @Column(name = "display_name")
    private String displayName;

    @Column(name = "distance_type")
    private String distanceType;

    @Column(name = "min_pace_sec_per_km")
    private int minPaceSecPerKm;

    @Column(name = "max_pace_sec_per_km")
    private int maxPaceSecPerKm;

    @Column(name = "image_url")
    private String imageUrl; // 이미지 파일 경로 또는 URL 추가

}

File: frontend_test/src/types/tier.ts
// src/types/tier.ts

/**
 * 티어 평가 API 요청 본문 타입
 */
export interface TierEvaluationRequest {
  distanceType: '5k' | '10k'; // 거리 유형
  recordId: number; // 기록 ID
}

/**
 * 티어 평가 API 응답 데이터 타입
 */
export interface TierData {
  tierId: number; // 티어 ID
  name: string; // 티어 영문명 (e.g., "SHOES")
  displayName: string; // 티어 한글명 (e.g., "구두")
  distanceType: '5k' | '10k'; // 거리 유형
  minPaceSecPerKm: number; // 최소 페이스 (km당 초)
  maxPaceSecPerKm: number; // 최대 페이스 (km당 초)
}
